sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_client_manager.py

```diff
@@ -380,8 +425,14 @@ def _setup_failover_endpoints(self):
                             **self._args
                         )
                     )
-        self._replica_clients = new_clients
+        random.shuffle(discovered_clients)
```

Comment:
We always shuffle here and sometimes we shuffle again at line 435. Can we avoid shuffling discovered_clients twice?